marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -652,14 +658,24 @@ private RunStepResponse runIngester(RunStepResponse runStepResponse, Collection<
     }
 
     private void processCsv(JacksonHandle jacksonHandle, File file) {
-        String uri = file.getParent();
-        if(SystemUtils.OS_NAME.toLowerCase().contains("windows")){
-            uri = "/" + FilenameUtils.separatorsToUnix(StringUtils.replaceOnce(uri, ":", ""));
+        String uri;
```

Comment:
I frequently like to put logic like this into a protected method so that it can easily unit-tested. You can even pass in the "osName" so you can verify the windows-specific code while not on windows. I didn't check the test coverage of all the conditions in here, but it'd be really easy to get 100% coverage of it with unit tests that will of course run very quickly too (i.e. put them in a class that doesn't extend anything, as you don't need any of the HubTestBase plumbing).

Reply:
Moved uri generation to a protected method and added test in WriteTestBatcherTest class